websites/C/Coursera/presence.ts

```diff
@@ -0,0 +1,350 @@
+					{
+						label: "View Course",
+						url: (() => {
+							return document.location.href.split("/").slice(0, 5).join("/");
+						})(),
+					},
+				],
+			},
+
+			{
+				path: /^\/learn\/.*\/?$/,
+				smallImageKey: () => Assets.Viewing,
+				smallImageText: () => "Viewing",
+				details: () => `
+				Viewing a Course`,
+				state: () => {
+					return document.querySelector("[data-e2e='hero-title']").textContent;
+				},
+				buttons: () => [{ label: "Course Link", url: document.location.href }],
+			},
+			{
+				path: /^\/specializations\/.+$/,
+				smallImageKey: () => Assets.Viewing,
+				smallImageText: () => "Viewing",
+				details: () => `
+				Viewing a Course`,
+				state: () => {
+					return document.querySelector("[data-e2e='hero-title']").textContent;
+				},
+				buttons: () => [{ label: "Course Link", url: document.location.href }],
+			},
+			{
+				path: /^\/search*/,
+				smallImageKey: () => Assets.Search,
+				smallImageText: () => "Searching",
+				details: () => {
+					return `searching for "${new URL(
+						document.location.href
+					).searchParams.get("query")}"`;
+				},
+
+				state: () => {
+					return `Found ${
+						document
+							.querySelector("[data-e2e='NumberOfResultsSection']")
+							.textContent.split(" ")[0]
+					} results`;
+				},
+				buttons: () => [{ label: "Results", url: document.location.href }],
+			},
+			{
+				path: /^\/+.*$/,
+				details: () => {
+					const currentUrl = document.location.href;
+					if (currentUrl.includes("my-learning")) return "My Courses";
+					else
+						return currentUrl.split("/")[3].toUpperCase().split("-").join(" ");
+				},
+				smallImageKey: () => Assets.Reading,
+				smallImageText: () => "Browsing",
+			},
+		];
+
+		return routes.find(route => route.path.test(path));
+	};
+
+presence.on("UpdateData", async () => {
+	const [showTimestamp, showButtons] = await Promise.all([
+			presence.getSetting<boolean>(Settings.TIMESTAMP),
+			presence.getSetting<boolean>(Settings.BUTTONS),
+		]),
+		presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+		};
+
+	if (showTimestamp) presenceData.startTimestamp = startTimestamp;
+
+	const route = router({
+		presenceData,
+		path: document.location.href.replace(
+			`https://${document.location.hostname}`,
+			""
+		),
+	});
+
+	if (!route) return presence.setActivity(presenceData);
+	if (route.state) presenceData.state = route.state();
+	if (route.details) presenceData.details = route.details();
+	if (showButtons && route.buttons) presenceData.buttons = route.buttons();
+	if (route.largeImageKey) presenceData.largeImageKey = route.largeImageKey();
+	if (route.smallImageKey) presenceData.smallImageKey = route.smallImageKey();
+	if (route.smallImageText)
+		presenceData.smallImageText = route.smallImageText();
+
+	if (showTimestamp && route.endTimestamp)
+		presenceData.endTimestamp = route.endTimestamp();
+
+	presence.setActivity(presenceData, route.playback ? route.playback() : false);
```

Comment:
You can remove the playback, the boolean is deprecated, and you don't use it anyway.

Reply:
no it's used

Reply:
Playback parameter is deprecated.
```suggestion
	presence.setActivity(presenceData);
```